File: backend/src/main/java/com/bigshen/blog/service/CategoryService.java
package com.bigshen.blog.service;

import com.bigshen.blog.dto.CategoryDTO;
import com.bigshen.blog.dto.CategoryRequest;
import com.bigshen.blog.model.Category;
import com.bigshen.blog.repository.CategoryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

/**
 * 分类服务
 */
@Service
@RequiredArgsConstructor
public class CategoryService {

    private final CategoryRepository categoryRepository;

    /**
     * 获取所有分类
     */
    public List<CategoryDTO> getAllCategories() {
        return categoryRepository.findAll()
                .stream()
                .map(this::toDTO)
                .collect(Collectors.toList());
    }

    /**
     * 根据ID获取分类
     */
    public CategoryDTO getCategoryById(Long id) {
        Category category = categoryRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("分类不存在"));
        return toDTO(category);
    }

    /**
     * 创建分类
     */
    public CategoryDTO createCategory(CategoryRequest request) {
        if (categoryRepository.existsByName(request.getName())) {
            throw new RuntimeException("分类名称已存在");
        }
        Category category = Category.builder()
                .name(request.getName())
                .description(request.getDescription())
                .articleCount(0)
                .build();
        category = categoryRepository.save(category);
        return toDTO(category);
    }

    /**
     * 更新分类
     */
    public CategoryDTO updateCategory(Long id, CategoryRequest request) {
        Category category = categoryRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("分类不存在"));
        category.setName(request.getName());
        category.setDescription(request.getDescription());
        category = categoryRepository.save(category);
        return toDTO(category);
    }

    /**
     * 删除分类
     */
    public void deleteCategory(Long id) {
        if (!categoryRepository.existsById(id)) {
            throw new RuntimeException("分类不存在");
        }
        categoryRepository.deleteById(id);
    }

    /**
     * 转换为DTO
     */
    private CategoryDTO toDTO(Category category) {
        return CategoryDTO.builder()
                .id(category.getId())
                .name(category.getName())
                .description(category.getDescription())
                .articleCount(category.getArticleCount())
                .build();
    }
}

File: frontend/src/services/categoryService.ts
import api from './api';
import type { Category } from '../types/article';

/**
 * 分类相关API
 */
export const categoryService = {
    getAllCategories: async (): Promise<Category[]> => {
        const response = await api.get<Category[]>('/categories');
        return response.data;
    },

    getCategoryById: async (id: number): Promise<Category> => {
        const response = await api.get<Category>(`/categories/${id}`);
        return response.data;
    },

    createCategory: async (data: { name: string; description: string }): Promise<Category> => {
        const response = await api.post<Category>('/categories', data);
        return response.data;
    },

    updateCategory: async (id: number, data: { name: string; description: string }): Promise<Category> => {
        const response = await api.put<Category>(`/categories/${id}`, data);
        return response.data;
    },

    deleteCategory: async (id: number): Promise<boolean> => {
        await api.delete(`/categories/${id}`);
        return true;
    },
};
